var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -176,37 +188,68 @@ class Qt(Package):
     conflicts("%apple-clang@13:", when="@:5.13")
 
     # Build-only dependencies
-    depends_on("pkgconfig", type="build")
+    for plat in ["linux", "darwin", "freebsd"]:
+        with when(f"platform={plat}"):
+            depends_on("pkgconfig", type="build")
+            depends_on("libsm", when="@3")
+            depends_on("glib", when="@4:")
+            depends_on("libmng")
+            depends_on("assimp@5.0.0:5", when="@5.5:+opengl")
+            depends_on("sqlite+column_metadata", when="+sql", type=("build", "run"))
+            depends_on("inputproto", when="@:5.8")
+    for plat in ["linux", "freebsd"]:
+        with when(f"platform={plat} +gui"):
+            depends_on("fontconfig")
+            depends_on("libsm")
+            depends_on("libx11")
+            depends_on("libxcb")
+            depends_on("libxkbcommon")
+            depends_on("xcb-util-image")
+            depends_on("xcb-util-keysyms")
+            depends_on("xcb-util-renderutil")
+            depends_on("xcb-util-wm")
+            depends_on("libxext")
+            depends_on("libxrender")
+
+        conflicts("+framework", msg="QT cannot be built as a framework except on macOS.")
```

Comment:
This conflict moved from an "only-macos" context into a "also-on-other-platforms" context: that should be restored.

Reply:
This moved from an "everywhere but MACOS" to an "everywhere but MACOS and Windows"